state-sync/aptos-data-client/src/aptosnet/mod.rs

```diff
@@ -503,95 +537,163 @@ impl DataSummaryPoller {
-                            .message("Peers to poll is empty!"))
+                            .message("No prioritized or regular peers to poll this round!"))
                     );
                 );
+                continue;
             }
 
-            // Go through each peer and poll individually
-            for peer in peers_to_poll {
-                // Start the peer polling timer
-                let timer = start_timer(
-                    &metrics::REQUEST_LATENCIES,
-                    StorageServiceRequest::GetStorageServerSummary
-                        .get_label()
-                        .into(),
+            // Go through each peer and poll them individually
+            if let Some(prioritized_peer) = prioritized_peer {
+                poll_peer(
+                    self.data_client.clone(),
+                    prioritized_peer,
+                    self.runtime.clone(),
                 );
+            }
+            if let Some(regular_peer) = regular_peer {
+                poll_peer(self.data_client.clone(), regular_peer, self.runtime.clone());
+            }
+        }
+    }
 
-                // Fetch the storage summary for the peer
-                let result: Result<StorageServerSummary> = self
-                    .data_client
-                    .send_request_to_peer_and_decode(
-                        peer,
-                        StorageServiceRequest::GetStorageServerSummary,
-                    )
-                    .await
-                    .map(Response::into_payload);
-                drop(timer);
-
-                // Check the storage summary response
-                let storage_summary = match result {
-                    Ok(storage_summary) => storage_summary,
-                    Err(error) => {
-                        error!(
-                            (LogSchema::new(LogEntry::StorageSummaryResponse)
-                                .event(LogEvent::PeerPollingError)
-                                .message("Error encountered when polling peer!")
-                                .error(&error)
-                                .peer(&peer))
-                        );
-                        continue;
-                    }
-                };
+    /// Fetches the next prioritized peer to poll or None
+    /// if an error was encountered.
+    fn fetch_prioritized_peer(&self) -> Option<PeerNetworkId> {
+        self.data_client
+            .fetch_prioritized_peer_to_poll()
+            .unwrap_or_else(|error| {
+                log_poller_error(error);
+                None
+            })
+    }
 
-                // Update the global storage summary and the summary for the peer
-                self.data_client.update_summary(peer, storage_summary);
-                self.data_client.update_global_summary_cache();
+    /// Fetches the next regular peer to poll based on the sample frequency
+    fn fetch_regular_peer(&self, always_poll: bool) -> Option<PeerNetworkId> {
```

Comment:
Should we keep the sampling logic anymore? Now that you just fetch one priority peer, should we always try to get some regular peers as well?

Reply:
I think the sampling still makes sense: we want to target priority peers in the happy path and only poll regular peers infrequently or (in the unhappy path) when we have no priority peers. I don't see a benefit in always polling regular peers in the happy-path, it just seems like unnecessary work.